src/index.ts

```diff
@@ -14,16 +14,14 @@ const run = async (args: IArgs) => {
   });
 
   client.on("message", async (message) => {
-    const botWasMention = message.mentions.users.find(
-      (u) => u.id == botUser.id
-    );
-    if (botWasMention && !message.author.bot) {
-      let messageContent = message.content.split(' ');
-      messageContent.shift()
-      const newMessage = messageContent.join(' ');
-      const service = await Dispatcher(newMessage);
+    const body = message.content;
+    console.log(body)
+    if (body.startsWith(`<@!${botUser.id}>`) && !message.author.bot) {
+      let components = body.split(" ");
+      components[0] = botUser.username
+      const service = await Dispatcher(components);
       if (!service) return;
```

Comment:
Rename this variable and remove the null/undefined check as this can never be null or undefined I believe

Reply:
Renamed it to "response" and removed the null/undefined check